blockchain/filedao/filedao_v2_manager.go

```diff
@@ -90,6 +90,24 @@ func (fm *FileV2Manager) FileDAOByHeight(height uint64) FileDAO {
 	return nil
 }
 
+// AddFileDAO add a new v2 file
+func (fm *FileV2Manager) AddFileDAO(fd *fileDAOv2, start uint64) error {
+	// update current top's end
+	top := fm.Indices[len(fm.Indices)-1]
+	end, err := top.fd.Height()
+	if err != nil {
+		return err
+	}
+	if start != end+1 {
+		// new file's first block != current tip height
+		return ErrInvalidTipHeight
```

Comment:
check and add next new v2 file, for example:
current top file start = 1000, end = 1500, then only `AddFileDAO(1501)` will be accepted